### PR TITLE
#6853 Fix crash when using an unescaped template

### PR DESCRIPTION
**Pull request by MV88**
## Description

The problem was related to the name of the feature attributes, especially if they contained a - like `${ properties.data-name }`
the crash happened inside template of lodash here
https://github.com/geosolutions-it/MapStore2/blob/b7b366d7d35491f659e4946dec800faff7cca887/web/client/components/data/identify/viewers/TemplateViewer.jsx#L19
So what i did is to prevent the crash with a try catch and to use square bracket notation if it finds a minus inside templates segments

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#6853

**What is the new behavior?**

no crash
the minus properties are managed correctly even if already escaped

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/utils/TemplateUtils.js [hunk 1]
```diff
@@ -33,7 +33,16 @@ export const validateStringAttribute = (feature, attribute, start = 0, end = 0)
 export const getCleanTemplate = (chosenTemplate, feature, regex, start = 0, end = 0, getDefaultMissingProperty = () => "") => {
     const matchVariables = isString(chosenTemplate) && chosenTemplate.match(regex);
     const replacedTag = matchVariables && matchVariables.map(temp => {
-        const varReplaced = temp.replace(/(<([^>]+)>)/ig, '');
+        let varReplaced = temp.replace(/(<([^>]+)>)/ig, '');
+        if (varReplaced.indexOf("-") !== -1) {
```
Comment (offtherailz) [inline web/client/utils/TemplateUtils.js:37]: I should not do this, but improve error messages and add a suggestion about the correct syntax.

File: web/client/components/data/identify/viewers/TemplateViewer.jsx [hunk 2]
```diff
@@ -14,10 +14,19 @@ import HtmlRenderer from '../../../misc/HtmlRenderer';
...
-                <HtmlRenderer html={template(getCleanTemplate(layer.featureInfo && layer.featureInfo.template || '', feature, /\$\{.*?\}/g, 2, 1))(feature)}/>
-            </div>
+        {response.features.map((feature, i) => {
+            const cleanTemplate = getCleanTemplate(layer.featureInfo && layer.featureInfo.template || '', feature, /\$\{.*?\}/g, 2, 1);
+            let html = "";
+            try {
+                html = template(cleanTemplate)(feature);
+            } catch (e) {
+                console.error(e);
+                html = "There was an error parsing the template, please check if the template contains a minus -, like ${ properties.data-array } ";
```
Comment (offtherailz) [inline web/client/components/data/identify/viewers/TemplateViewer.jsx:24]: localize this message

**Review by offtherailz — CHANGES_REQUESTED** [review]
I suggest to 
- [ ] improve the hints about the correct syntax to use in the template. adding `${properties.id}, the name is ${properties['attribute-name'] 
image

- [ ] show an error in case of usage of `properties.attribute-name`.

This helps in the future to support `-` for operations, for instance, instead of supporting names like `properties.attribute-name` that are not compliant with any template language.

The correct syntax to use is `properties['attribute-name']`
 @giohappy

File: web/client/translations/data.it-IT.json [hunk 3]
```diff
@@ -102,7 +102,9 @@
             "styleWarning": "L'editore dello stile di livello per il tipo di geometria '{geometryType}' non è supportato.",
             "templateFormatInfoAlert1": "Click sul bottone di modifica per aggiungere un nuovo template.",
             "templateFormatInfoAlert2": "Usa ${ attribute } per aggiungere gli attributi da visualizzare",
-            "templateFormatInfoAlertExample": "id della feature è ${ properties }",
+            "templateFormatInfoAlertExample": "id della feature è ${ properties } o",
+            "templateFormatInfoAlertExample2": "{ properties['nome-cognome'] }",
+            "templateError": "Si è verificato un errore durante l'applicazione del formato del modello alle informazioni restituite dal server, controlla il formato del modello",
```
Comment (offtherailz) [inline web/client/translations/data.it-IT.json:107]:
```suggestion
            "templateError": "Si è verificato un errore durante l'applicazione del template alle informazioni restituite dal server, controlla il template.",
```

File: web/client/translations/data.en-US.json [hunk 4]
```diff
@@ -102,7 +102,9 @@
             "styleWarning": "Layer style editor for geometry type '{geometryType}' is not supported.",
             "templateFormatInfoAlert1": "Click on edit button to add a new template.",
             "templateFormatInfoAlert2": "Use ${ attribute } to wrap the properties you need to display",
-            "templateFormatInfoAlertExample": "The id of the feature is ${ properties }",
+            "templateFormatInfoAlertExample": "The id of the feature is ${ properties } or",
+            "templateFormatInfoAlertExample2": "{ properties['data-value'] }",
+            "templateError": "There was an error applying the template format to the information returned by the server, please check the template",
```
Comment (offtherailz) [inline web/client/translations/data.en-US.json:107]:
```suggestion
            "templateError": "There was an error applying the template to the information returned by the server, please check the template",
```

File: web/client/translations/data.es-ES.json [hunk 5]
```diff
@@ -141,7 +141,9 @@
               "legendHeight": "Altura",
               "legendPreview": "Avance"
             },
-            "templateFormatInfoAlertExample": "La identificación de la característica es ${ properties }",
+            "templateFormatInfoAlertExample": "La identificación de la característica es ${ properties } ora",
+            "templateFormatInfoAlertExample2": "{ properties['datos-valor'] }",
+            "templateError": "Hubo un error al aplicar el formato de la plantilla a la información devuelta por el servidor, verifique la plantilla",
```
Comment (offtherailz) [inline web/client/translations/data.es-ES.json:146]:
```suggestion
            "templateError": "Hubo un error al aplicar la plantilla a la información devuelta por el servidor, verifique la plantilla",
```

File: web/client/translations/data.fr-FR.json [hunk 6]
```diff
@@ -102,7 +102,9 @@
             "styleWarning": "L'éditeur de style de calque pour le type de géométrie '{geometryType}' n'est pas pris en charge.",
             "templateFormatInfoAlert1": "Cliquez sur le bouton d'édition pour ajouter un nouveau modèle.",
             "templateFormatInfoAlert2": "Utilisez ${ attribut } pour envelopper les propriétés que vous devez afficher",
-            "templateFormatInfoAlertExample": "L'ID de la fonctionnalité est ${ properties }",
+            "templateFormatInfoAlertExample": "L'ID de la fonctionnalité est ${ properties } ou",
+            "templateFormatInfoAlertExample2": "{ properties['données-valeur'] }",
+            "templateError": "Une erreur s'est produite lors de l'application du format du modèle aux informations renvoyées par le serveur, veuillez vérifier le modèle",
```
Comment (offtherailz) [inline web/client/translations/data.fr-FR.json:107]:
```suggestion
            "templateError": "Une erreur s'est produite lors de l'application du modèle aux informations renvoyées par le serveur, veuillez vérifier le modèle",
```

File: web/client/translations/data.de-DE.json [hunk 7]
```diff
@@ -102,7 +102,9 @@
             "styleWarning": "Ebenenstil-Editor für Geometrietyp '{geometryType}' wird nicht unterstützt.",
             "templateFormatInfoAlert1": "Klicken Sie auf Bearbeiten, um eine neue Vorlage hinzuzufügen.",
             "templateFormatInfoAlert2": "Verwenden Sie ${ attribute }, um Attribut-Werte anzuzeigen",
-            "templateFormatInfoAlertExample": "Die ID des Features lautet ${ properties }",
+            "templateFormatInfoAlertExample": "Die ID des Features lautet ${ properties } oder",
+            "templateFormatInfoAlertExample2": "{ properties['daten-wert'] }",
+            "templateError": "Beim Anwenden des Vorlagenformats auf die vom Server zurückgegebenen Informationen ist ein Fehler aufgetreten. Bitte überprüfen Sie die Vorlage",
```
Comment (offtherailz) [inline web/client/translations/data.de-DE.json:107]:
```suggestion
            "templateError": "Beim Anwenden der Vorlage auf die vom Server zurückgegebenen Informationen ist ein Fehler aufgetreten. Bitte überprüfen Sie die Vorlage",
```

**Review by offtherailz — APPROVED** [review]
The code is ok for me. Anyway I can not replicate the error message. Please check if the error message is effectively shown.
Approved by my side.

**Comment by tdipisa** [comment]
@ElenaGallo please test this in DEV and let us know if we can backport.

**Comment by ElenaGallo** [comment]
@MV88 test passed, please backport this PR to the stable branch

**Pull request merged**